### ISSUE TITLE
Simultaneous requests share the same state

### ISSUE THREAD

**Issue by Vest** [SAP/odata-library#35]
Hello everyone,

when I was trying to use the library in Node.js 14.16.1, I have found out that somewhere deep in the library, the shared object is used. This doesn't allow me to send simultaneous requests properly. It seems that the state is shared by subsequent requests, and when the async request is processed by node, my web service always gets the same payload. Here is an example:

```js
const Service = require("@sap_oss/odata-library").Service;
const SERVICE_URL = "https://services.odata.org/V4/OData/OData.svc/";

(async () => {
    const sample = new Service(SERVICE_URL);
    await sample.init;

    console.log(`${SERVICE_URL} initialized successfully`);

    const results = await Promise.all([
        sample.Products.select(["ID", "Name"]).get(0),
        sample.Products.select(["ID", "Name"]).get(1)
    ]);

    console.log(`Results ¯\\_(ツ)_/¯ ${JSON.stringify(results)}`);
})();
```

Here is my result:
```https://services.odata.org/V4/OData/OData.svc/ initialized successfully
Results ¯\_(ツ)_/¯ [[{"ID":0,"Name":"Bread"}],[{"ID":0,"Name":"Bread"}]]
```

Honestly, I have expected two different Products.

When I debugged your code, I have found out that the issue is caused by the library design. Somewhere in `QueryableResource.js`, the object `defaultRequest` is shared.

I can use await, it will make my code sequential, this is not what I want. Or, I can initialize a service asynchronously per each task/request, but this is also clumsy, because every `init` call reads $metadata.

None of the provided examples demonstrate my case, that is why I had to improvise. I hope this bug report will be helpful for other users, if they have similar issues.

p.s. it was difficult to find, why my code didn't work.

**Comment by norbertvolf** [comment]
Hi,

You are right. We have been using the library for testing with sequential reading normally. Thank you for issue.

BTW. I have fixed the bug with double slash which have you sent me via comment in [PR](https://github.com/SAP/odata-library/pull/1/commits/731f7f74caf404806a9d6384f1e442ab5e109c29). The fix is in version 0.9.14.

**Comment by Vest** [comment]
Thanks, I have noticed this. Appreciate the quick fix.
I was really disappointed, when I faced this issue, my thoughts were "how that was ever possible, that nobody faced this bug in last 8 months (that commit date)".

**Comment by norbertvolf** [comment]
Hi @Vest,

I have published version [0.10.0](https://www.npmjs.com/package/@sap_oss/odata-library/v/0.10.0) with patch which solves the issue with simultaneous requests.
